### PR TITLE
SLS-63 Extend dhandle refresh mechanism to support multiple tables

### PR DESCRIPTION
**Pull request by korteland**
Pretty much exactly as described on the ticket. Thanks Peter!

### CODE REVIEW

File: src/conn/conn_oligarch.c [hunk 1]
```diff
@@ -309,19 +310,21 @@ __wt_oligarch_manager_remove_table(WT_SESSION_IMPL *session, uint32_t ingest_id)
...
+    conn = S2C(session);
+    entry = conn->oligarch_manager.entries[ingest_id];
 
     *cursorp = NULL;
 
-    if (manager->update_dhandle) {
-        manager->update_dhandle = false;
+    WT_ACQUIRE_READ(global_ckpt_id, conn->disaggregated_storage.global_checkpoint_id);
+    if (global_ckpt_id > entry->global_checkpoint_id) {
+        entry->global_checkpoint_id = global_ckpt_id;
```
Comment (quchenhao) [inline src/conn/conn_oligarch.c:327]: Should we update this only after we successfully refreshed?

Comment (quchenhao) [reply src/conn/conn_oligarch.c:327]: This is also a shared variable?

Comment (korteland) [reply src/conn/conn_oligarch.c]: To your first point, that's valid - I reorganised the code a little. For Peter's benefit - we discussed this offline, and this is only used by the log replay code so it's a single-threaded context.

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/include/connection.h [hunk 2]
```diff
@@ -127,6 +127,7 @@ struct __wt_oligarch_manager_entry {
 
     uint64_t accumulated_write_bytes;
     uint64_t checkpoint_txn_id;
+    uint64_t global_checkpoint_id;
```
Comment (quchenhao) [inline src/include/connection.h:130]: Can we rename this to known_global_checkpoint_id?

Comment (korteland) [reply src/include/connection.h:130]: Hmm I agree a rename makes sense but `known` doesn't quite convey what it does. What about just `read_checkpoint`?

Comment (quchenhao) [reply src/include/connection.h:130]: sgtm

Comment (korteland) [reply src/include/connection.h]: Done in bf3990e5d

**Review by pmacko86 — APPROVED** [review]
LGTM. Very nice!

**Pull request merged**